client/js/widgets/button.js

```diff
@@ -255,61 +250,68 @@ export class Button extends Widget {
           case 'startsWith':
           case 'toLocaleLowerCase':
           case 'toLocaleUpperCase':
-            variables[v] = x[a.operation](y);
+            v = x[o](y);
             break;
           case 'replace':
           case 'replaceAll':
           case 'substr':
-            variables[v] = x[a.operation](y, z);
+            v = x[o](y, z);
             break;
 
           // Array operations
           // 'length' should work the same as for strings
           case 'getIndex':
-            variables[v] = x[y];
+            v = x[y];
             break;
           case 'setIndex':
-            variables[v][x] = y;
+            v[x] = y;
             break;
           case 'from':
           case 'isArray':
-            variables[v] = Array[a.operation](x);
+            v = Array[o](x);
             break;
           case 'concatArray':
-            variables[v] = x.concat(y);
+            v = x.concat(y);
             break;
           case 'pop':
           case 'reverse':
           case 'shift':
           case 'sort':
-            variables[v] = x[a.operation]();
+            v = x[o]();
             break;
           case 'findIndex':
           case 'includes':
           case 'indexOf':
           case 'join':
           case 'lastIndexOf':
-            variables[v] = x[a.operation](y);
+            v = x[o](y);
             break;
           case 'slice':
-            variables[v] = x[a.operation](y, z);
+            v = x[o](y, z);
             break;
           case 'push':
           case 'unshift':
-            variables[v][a.operation](x);
+            if (Array.isArray(v)) v[o](x);
```

Comment:
Why do you add a check here but nowhere else? Many other operations should have the same problem which should already be caught and added to `problems` anyway.